### PR TITLE
chore: Add ReleaseNotesBuilder.

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

Add ReleaseNotesBuilder.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: build/ReleaseNotesBuilder/data.yml
Comment (chynesNR) [inline build/ReleaseNotesBuilder/data.yml:1]: Nice, I like this approach

File: build/ReleaseNotesBuilder/Program.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        {
+            var result = Parser.Default.ParseArguments<Options>(args)
+                .WithParsed(ValidateOptions)
+                .WithNotParsed(HandleParseError);
+
+            var persistentData = LoadPersisentData(result.Value.PersistentData);
+            var checksums = LoadChecksums(result.Value.Checksums);
+            var maker = new ReleaseNotesModel(persistentData, checksums);
+            var changelog = File.ReadAllLines(result.Value.Changelog).ToList();
+            var releaseVersion = ChangelogParser.Parse(changelog, maker);
```
Comment (chynesNR) [inline build/ReleaseNotesBuilder/Program.cs:26]: Could we get a try/catch at least around the Parse call? With all the string manipulation it'd be easy for an error to sneak in somewhere down the road.

**Comment by chynesNR** [comment]
Looking good!

File: build/ReleaseNotesBuilder/ReleaseNotesModel.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+            _subject = persistentData.Subject;
+            _releaseDate = DateTime.Now.ToString("yyyy-MM-dd");
+            _downloadLink = persistentData.DownloadLink;
+            _preamble = persistentData.Preamble;
+            _epilogue = persistentData.Epilogue;
+            _checksums = checksums;
+        }
+
+        public string Make()
```
Comment (nrcventura) [inline build/ReleaseNotesBuilder/ReleaseNotesModel.cs:53]: A more descriptive name for this method might make it more self-documenting. Maybe something like `GetDocumentContents`, `GetFormattedContents`, or `FormatForDocsSite`?

Comment (jaffinito) [reply build/ReleaseNotesBuilder/ReleaseNotesModel.cs:53]: Will do!

File: build/ReleaseNotesBuilder/Program.cs (hunk 1 above)
Comment (jaffinito) [reply build/ReleaseNotesBuilder/Program.cs:26]: Wrapped several methods in try-catches.

**Review by nrcventura — APPROVED** [review]

**Review by tippmar-nr — APPROVED** [review]

**Pull request merged**